x/oracle/keeper/oracle.go

```diff
@@ -42,6 +43,53 @@ func (k Keeper) RegisterOracle(ctx sdk.Context, msg *types.MsgRegisterOracle) er
 	return nil
 }
 
+func (k Keeper) UpdateOracleInfo(ctx sdk.Context, msg *types.MsgUpdateOracleInfo) error {
+	oracle, err := k.GetOracle(ctx, msg.OracleAddress)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+	}
+
+	blockTime := ctx.BlockHeader().Time
+
+	if !oracle.OracleCommissionRate.Equal(msg.OracleCommissionRate) {
+		if err := k.ValidateOracleCommission(oracle, blockTime, msg.OracleCommissionRate); err != nil {
+			return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+		}
+	}
+
+	oracle.UpdateTime = blockTime
+	oracle.Endpoint = msg.Endpoint
+	oracle.OracleCommissionRate = msg.OracleCommissionRate
+
+	if err := k.SetOracle(ctx, oracle); err != nil {
+		return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+	}
+	return nil
+}
+
+// ValidateOracleCommission validate an oracle's commission rate.
+// An error is returned if the new commission rate is invalid.
+func (k Keeper) ValidateOracleCommission(oracle *types.Oracle, blockTime time.Time, newRate sdk.Dec) error {
+	switch {
+	case blockTime.Sub(oracle.UpdateTime).Hours() < 24:
+		return types.ErrCommissionUpdateTime
+
+	case newRate.IsNegative():
+		// new rate cannot be negative
+		return types.ErrCommissionNegative
+
+	case newRate.GT(oracle.OracleCommissionMaxRate):
+		// new rate cannot be greater than the max rate
+		return types.ErrCommissionGTMaxRate
+
+	case newRate.Sub(oracle.OracleCommissionRate).GT(oracle.OracleCommissionMaxChangeRate):
```

Comment:
I think you need to change like below:
```suggestion
	case newRate.Sub(oracle.OracleCommissionRate).Abs().GT(oracle.OracleCommissionMaxChangeRate):
```

Reply:
This part requires consideration of limitations when reducing the commission rate. In staking module, there is no limitation about this. 
In my opinion, when the commission rate is reduced, the provider does not lose money, so it would be okay to give this part as freedom.

Reply:
Oh, I didn't know that staking module only restrict max-change-rate when increase the commission. As @audtlr24 said, it seems good to follow the way the staking module do.